source/slang/compiler.cpp

```diff
@@ -416,18 +416,63 @@ namespace Slang
         return result;
     }
 
+    void reportExternalCompileError(const char* compilerName, SlangResult res, const UnownedStringSlice& diagnostic, DiagnosticSink* sink)
+    {
+        StringBuilder builder;
+        if (compilerName)
+        {
+            builder << compilerName << ": ";
+        }
+
+        if (diagnostic.size() > 0)
+        {
+            builder.Append(diagnostic);
+        }
+
+        if (SLANG_FAILED(res) && res != SLANG_FAIL)
+        {
+            {
+                char tmp[17];
+                sprintf_s(tmp, SLANG_COUNT_OF(tmp), "0x%08x", uint32_t(res));
+                builder << "Result(" << tmp << ") ";
+            }
+
+            PlatformUtil::appendResult(res, builder);
+        }
+
+        // TODO(tfoley): need a better policy for how we translate diagnostics
+        // back into the Slang world (although we should always try to generate
+        // HLSL that doesn't produce any diagnostics...)
+        sink->diagnoseRaw(SLANG_FAILED(res) ? Severity::Error : Severity::Warning, builder.getUnownedSlice());
+    }
+
 #if SLANG_ENABLE_DXBC_SUPPORT
-   
-    List<uint8_t> EmitDXBytecodeForEntryPoint(
+
+    static UnownedStringSlice _getSlice(ID3DBlob* blob)
+    {
+        if (blob)
+        {
+            const char* chars = (const char*)blob->GetBufferPointer();
+            size_t len = blob->GetBufferSize();
+            len -= size_t(len > 0 && chars[len - 1] == 0);
+            return UnownedStringSlice(chars, len);
+        }
+        return UnownedStringSlice();
+    }
+
+    SlangResult emitDXBytecodeForEntryPoint(
         EntryPointRequest*  entryPoint,
-        TargetRequest*      targetReq)
+        TargetRequest*      targetReq,
+        List<uint8_t>&      byteCodeOut)
```

Comment:
I think the coding convention is for `outByteCode` instead of `byteCodeOut`, but as always I'm not planning to make a fuss about it.